server/sonar-server/src/main/java/org/sonar/server/qualityprofile/index/ActiveRuleIndex.java

```diff
@@ -19,179 +19,211 @@
+  /**
+   * @deprecated since 5.5, use {@link org.sonar.db.qualityprofile.ActiveRuleDao instead}
+   */
+  @Deprecated
+  @CheckForNull
+  public ActiveRule getNullableByKey(ActiveRuleKey key) {
+    GetRequestBuilder request = getClient().prepareGet()
+      .setIndex(RuleIndexDefinition.INDEX)
+      .setType(RuleIndexDefinition.TYPE_ACTIVE_RULE)
+      .setId(key.toString())
+      .setFetchSource(true)
+      .setRouting(key.ruleKey().repository());
+
+    GetResponse response = request.get();
+    if (response.isExists()) {
+      return new ActiveRuleDoc(response.getSource());
     }
-    return mapping;
-  }
-
-  @Override
-  protected Map mapDomain() {
-    Map<String, Object> mapping = new HashMap<>();
-    mapping.put("dynamic", false);
-    mapping.put("_id", mapKey());
-    mapping.put("_parent", mapParent());
-    mapping.put("_routing", mapRouting());
-    mapping.put("properties", mapProperties());
-    return mapping;
-  }
-
-  private Map mapRouting() {
-    Map<String, Object> mapping = new HashMap<>();
-    mapping.put("required", true);
-    mapping.put("path", ActiveRuleNormalizer.ActiveRuleField.RULE_KEY.field());
-    return mapping;
-  }
-
-  private Object mapParent() {
-    Map<String, Object> mapping = new HashMap<>();
-    mapping.put("type", this.getParentType());
-    return mapping;
-  }
-
-  @Override
-  public ActiveRule toDoc(Map<String, Object> fields) {
-    return new ActiveRuleDoc(fields);
+    return null;
   }
 
   /**
-   * finder methods
+   * @deprecated since 5.5, use {@link org.sonar.db.qualityprofile.ActiveRuleDao instead}
    */
+  @Deprecated
   public List<ActiveRule> findByRule(RuleKey key) {
-    SearchRequestBuilder request = getClient().prepareSearch(this.getIndexName())
+    SearchRequestBuilder request = getClient().prepareSearch(RuleIndexDefinition.INDEX)
       .setQuery(QueryBuilders
-        .hasParentQuery(this.getParentType(),
-          QueryBuilders.idsQuery(this.getParentType())
+        .hasParentQuery(RuleIndexDefinition.TYPE_RULE,
+          QueryBuilders.idsQuery(RuleIndexDefinition.TYPE_RULE)
             .addIds(key.toString())
         ))
-      .setRouting(key.toString())
-      // TODO replace by scrolling
+      .setRouting(key.repository())
       .setSize(Integer.MAX_VALUE);
 
     SearchResponse response = request.get();
 
     List<ActiveRule> activeRules = new ArrayList<>();
     for (SearchHit hit : response.getHits()) {
-      activeRules.add(toDoc(hit.getSource()));
+      activeRules.add(new ActiveRuleDoc(hit.getSource()));
     }
     return activeRules;
   }
 
-  public Iterator<ActiveRule> findByProfile(String key) {
-    SearchRequestBuilder request = getClient().prepareSearch(getIndexName())
-      .setTypes(IndexDefinition.ACTIVE_RULE.getIndexType())
+  /**
+   * @deprecated since 5.5, use {@link org.sonar.db.qualityprofile.ActiveRuleDao instead}
+   */
+  @Deprecated
+  public Iterator<ActiveRuleDoc> findByProfile(String key) {
```

Comment:
![INFO](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-info.png) Do not forget to remove this deprecated code someday. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1133)